okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UserAgentInterceptor.java

```diff
@@ -28,14 +31,31 @@
     private static final Pattern VALID_USER_AGENT = Pattern.compile("[A-Za-z0-9()\\-#;/.,_\\s]+");
     private final String userAgent;
 
-    private UserAgentInterceptor(String userAgent) {
+    private UserAgentInterceptor(String userAgent, Class<?> serviceClass) {
         Preconditions.checkArgument(VALID_USER_AGENT.matcher(userAgent).matches(),
                 "User Agent must match pattern '%s': %s", VALID_USER_AGENT, userAgent);
-        this.userAgent = userAgent;
+
+        String version = getVersionString(serviceClass);
+        this.userAgent = version.isEmpty()
+                ? userAgent
+                : String.format("%s (%s)", userAgent, version);
+    }
+
+    private String getVersionString(Class<?> serviceClass) {
+        List<String> versions = new ArrayList<>(2);
+        String maybeRemotingVersion = this.getClass().getPackage().getImplementationVersion();
+        if (maybeRemotingVersion != null) {
+            versions.add("http-remoting " + maybeRemotingVersion);
```

Comment:
To make analysis simpler we probably want this to come second.

Reply:
done